prov/verbs/src/verbs_info.c

```diff
@@ -1007,6 +1012,162 @@ static int vrb_ifa_rdma_info(const struct ifaddrs *ifa, char **dev_name,
 	return ret;
 }
 
+int vrb_get_port_space(const struct fi_info *info)
+{
+	if (info != NULL && info->addr_format == FI_SOCKADDR_IB)
+		return RDMA_PS_IB;
+	else
+		return RDMA_PS_TCP;
+}
+
+static int vrb_gid_is_null(const union ibv_gid *gid)
+{
+	return !(gid->raw[8]  | gid->raw[9]  | gid->raw[10] | gid->raw[11] |
+	         gid->raw[12] | gid->raw[13] | gid->raw[14] | gid->raw[15]);
+}
+
+static int vrb_gid_to_str(const union ibv_gid *gid, char gid_str[40])
+{
+	if (sprintf(gid_str, "%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x",
+		        gid->raw[ 0], gid->raw[ 1],
+		        gid->raw[ 2], gid->raw[ 3],
+		        gid->raw[ 4], gid->raw[ 5],
+		        gid->raw[ 6], gid->raw[ 7],
+		        gid->raw[ 8], gid->raw[ 9],
+		        gid->raw[10], gid->raw[11],
+		        gid->raw[12], gid->raw[13],
+		        gid->raw[14], gid->raw[15]) != 39)
+	{
+		return -FI_EINVAL;
+	}
+	return 0;
+}
```

Comment:
There are address printing functions available in the common code.  A GID can be converted using inet_ntop as ipv6

Reply:
I will modify the code to use `ofi_straddr()`.